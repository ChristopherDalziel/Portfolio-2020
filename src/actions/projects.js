import database from "../firebase/firebase";

export const createProject = (project) => ({
  type: "CREATE_PROJECT",
  project,
});

export const startCreateProject = (projectData = {}) => {
  return (dispatch, getState) => {
    const { name = "", description = "", technology = "" } = projectData;
    const project = { name, description, technology };
    return database
      .ref("projects")
      .push(project)
      .then((ref) => {
        dispatch(createProject({ id: ref.key, ...project }));
      });
  };
};
