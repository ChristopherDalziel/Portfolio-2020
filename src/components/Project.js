import React from "react";
import styled from "@emotion/styled";

const ProjectContainer = styled.div`
  border: solid 1px black;
  height: 25%
  width: 75%;
  padding: 50px;
  margin: 15px
`;

const ProjectImage = styled.img`
  height: 25%;
`;

const Project = ({ name, description, technology, url }) => (
  <ProjectContainer>
    <p>{name}</p>
    <p>{description}</p>
    <p>{technology}</p>
    <img height="250px" src={url} alt="related to project name" />
  </ProjectContainer>
);

export default Project;
