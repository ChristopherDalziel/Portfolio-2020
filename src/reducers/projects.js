const projectReducerDefaultState = [];

const projectReducer = (state = projectReducerDefaultState, action) => {
  switch (action.type) {
    case "CREATE_PROJECT":
      return [...state, action.project];
    case "SET_PROJECTS":
      return action.projects;
    default:
      return state;
  }
};

export default projectReducer;
