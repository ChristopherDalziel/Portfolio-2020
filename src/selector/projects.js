const getVisibleProjects = (projects, { text }) => {
  return projects.filter((project) => {
    const matchText =
      project.name.toLowerCase().includes(text.toLowerCase()) ||
      project.technology.toLowerCase().includes(text.toLowerCase());

    return matchText;
  });
};

export default getVisibleProjects;
