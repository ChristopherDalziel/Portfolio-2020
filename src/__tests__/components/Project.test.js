import React from "react";
import { shallow, mount, render } from "enzyme";
import toJSON from "enzyme-to-json";
import Project from "../../components/Project";
import projects from "../fixtures/projects";

test("Should render individual project with data", () => {
  const wrapper = shallow(
    <Project
      name={projects[2].name}
      description={projects[2].description}
      technology={projects[2].technology}
      githubUrl={projects[2].githubUrl}
      url={projects[2].url}
    />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
