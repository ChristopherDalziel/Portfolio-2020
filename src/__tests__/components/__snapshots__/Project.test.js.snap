// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render individual project with data 1`] = `
<Styled(div)
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="related to project name"
    src="Image Url"
    width="650px"
  />
   
</Styled(div)>
`;
